<a href="https://colab.research.google.com/github/26Sanya/Assignments-CiSc/blob/master/Assignment_2_(Graded).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# INSTRUCTIONS

Read the instructions metioned below carefully before attempting the assignment:

    - This assignment has 12 questions and all the questions are mandatory.
    - Read the desciption of the dataset mentioned below carefully.
    - Each question has been commented and written in an independent cell.
    - Write your code in the cell just below the question (Refer Ques-1 - It has been solved for you to understand)
    - Try to submit the assignment within the deadline.
    - Plagiarism will have a huge penality.
    - After completing the assignment upload this notebook to your github and submit the assignment on Google Classroom.
    
    
##### Download the Dataset file(.csv) from here:

https://drive.google.com/file/d/146KzhkfeuzDigbOjaZ1nkTpAHzp7TATz/view?usp=sharing


# COVID19 India dataset

### Description

Title of dataset: covid19india.csv

Sources:

    * A crowdsourced initiative - covid19india.org
    * Date received - 19/07/2020 
    
Number of Instances/Rows

    * Total Number of Rows = 381

Number of Attributes/Fearures/Columns

    * 2 Object, 39 Numerical attributes. 

Attribute Information/Fearures/Columns

    * date: Object Type
    * Status: Confirmed, Recovered, Deceased
    * Below mentioned are remaining 38 columns representing States and UT of India
    
    State 
    Maharashtra, Tamil Nadu, Delhi, Gujarat, Karnataka, Uttar Pradesh, Telangana, Andhra Pradesh, West Bengal, Rajasthan, Haryana, Bihar, Madhya Pradesh, Assam, Odisha, Jammu and Kashmir, Punjab, Kerala, Chhattisgarh, Jharkhand, Uttarakhand, Goa, State Unassigned, Tripura, Manipur, Puducherry, Himachal Pradesh, Ladakh, Nagaland, Chandigarh, Dadra and Nagar Haveli and Daman and Diu, Arunachal Pradesh, Meghalaya, Mizoram, Andaman and Nicobar Islands, Sikkim, Lakshadweep                       
    
    State_code
    MH, TN, DL, GJ, KA, UP, TG, AP, WB, RJ, HR, BR, MP, AS, OR, JK, PB, KL, CT, JH, UT, GA, UN, TR, MN, PY, HP, LA, NL, CH, DN, AR, ML, MZ, AN, SK, LD

In [1]:
# Q1 - Import the required Library (Sample question and answer)

import numpy as np
import pandas as pd

In [2]:
# Q2 - Read the csv file to the pandas dataframe
df= pd.read_csv('/content/covid19india.csv')

In [3]:
# Q3 - Display the top 5 rows of the dataframe
df.head()

date     status  an  ap  ar  as  br  ch  ...  sk  tg  tn  tr  un  up  ut  wb
0  14-Mar-20  Confirmed   0   1   0   0   0   0  ...   0   1   1   0   0  12   0   0
1  14-Mar-20  Recovered   0   0   0   0   0   0  ...   0   0   0   0   0   4   0   0
2  14-Mar-20   Deceased   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   0
3  15-Mar-20  Confirmed   0   0   0   0   0   0  ...   0   2   0   0   0   1   0   0
4  15-Mar-20  Recovered   0   0   0   0   0   0  ...   0   1   0   0   0   0   0   0

[5 rows x 40 columns]

In [4]:
# Q4 - Display the bottom 5 rows of the dataframe

df.tail()

date     status  an    ap  ar    as  ...    tn   tr   un    up   ut    wb
376  17-Jul-20  Recovered   0   905  22   551  ...  3391   47    0   959   26   838
377  17-Jul-20   Deceased   0    42   0     1  ...    79    0    0    38    1    26
378  18-Jul-20  Confirmed   0  3963  41  1117  ...  4807  171 -368  1873  174  2198
379  18-Jul-20  Recovered   0  1465  99  1060  ...  3049   51    0  1030   60  1286
380  18-Jul-20   Deceased   0    52   0     2  ...    88    2    0    24    1    27

[5 rows x 40 columns]

In [5]:
# Q5 - How many columns and rows are there in the dataframe?


Rows= 381 
<br> Columns= 40


In [6]:
# Q6 - Display a breif overview of all the columns data type and missing values?
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 381 entries, 0 to 380
Data columns (total 40 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   date    381 non-null    object
 1   status  381 non-null    object
 2   an      381 non-null    int64 
 3   ap      381 non-null    int64 
 4   ar      381 non-null    int64 
 5   as      381 non-null    int64 
 6   br      381 non-null    int64 
 7   ch      381 non-null    int64 
 8   ct      381 non-null    int64 
 9   dd      381 non-null    int64 
 10  dl      381 non-null    int64 
 11  dn      381 non-null    int64 
 12  ga      381 non-null    int64 
 13  gj      381 non-null    int64 
 14  hp      381 non-null    int64 
 15  hr      381 non-null    int64 
 16  jh      381 non-null    int64 
 17  jk      381 non-null    int64 
 18  ka      381 non-null    int64 
 19  kl      381 non-null    int64 
 20  la      381 non-null    int64 
 21  ld      381 non-null    int64 
 22  mh      381 non-null    in

In [7]:
# Q7 - Display the descriptive statistical analysis on numerical columns of the dataframe?

df.describe()

an           ap  ...          ut           wb
count  381.000000   381.000000  ...  381.000000   381.000000
mean     0.821522   175.742782  ...   19.446194   170.141732
std      2.422408   436.279518  ...   36.746792   314.410201
min      0.000000     0.000000  ...    0.000000   -19.000000
25%      0.000000     1.000000  ...    0.000000     3.000000
50%      0.000000    25.000000  ...    1.000000    18.000000
75%      0.000000    97.000000  ...   25.000000   188.000000
max     20.000000  3963.000000  ...  216.000000  2198.000000

[8 rows x 38 columns]

In [8]:
# Q8 - What were the number of confirmed cases in Rajasthan on 1st July 2020?
print(df.loc[(df['date']== '01-Jul-20') & (df['status']=='Confirmed'),['rj']])

      rj
327  304


Number of Confirmed cases in Rajasthan on 1st July 2020 = 304
which is listed at index 327 in our dataframe.

In [9]:
# Q9 - What is the number of Deceased cases on 18th July 2020?

print(df.loc[(df['date']== '18-Jul-20') & (df['status']=='Deceased')].sum(axis=1))

380    543
dtype: int64


The number of deceased cases on 18th July 2020 is sum of deceased in all states on that day i.e. row wise sum at index of 18-Jul-20/Deceased (380) = 543.

In [10]:
# Q10 - Create a new column 'total'. Values in this column is supposed to contain the sum of corresponding row.
df["total"] = df.sum(axis=1)
df.tail()

date     status  an    ap  ar    as  ...   tr   un    up   ut    wb  total
376  17-Jul-20  Recovered   0   905  22   551  ...   47    0   959   26   838  17476
377  17-Jul-20   Deceased   0    42   0     1  ...    0    0    38    1    26    676
378  18-Jul-20  Confirmed   0  3963  41  1117  ...  171 -368  1873  174  2198  37407
379  18-Jul-20  Recovered   0  1465  99  1060  ...   51    0  1030   60  1286  23552
380  18-Jul-20   Deceased   0    52   0     2  ...    2    0    24    1    27    543

[5 rows x 41 columns]

In [11]:
# Q11 - Group by on 'Status' column and display first.

grp= df.groupby('status')
grp.first()

date  an  ap  ar  as  br  ch  ...  tn  tr  un  up  ut  wb  total
status                                        ...                               
Confirmed  14-Mar-20   0   1   0   0   0   0  ...   1   0   0  12   0   0     81
Deceased   14-Mar-20   0   0   0   0   0   0  ...   0   0   0   0   0   0      2
Recovered  14-Mar-20   0   0   0   0   0   0  ...   0   0   0   4   0   0      9

[3 rows x 40 columns]

In [12]:
# Q12 - Display the overall basic statistical analysis of Rajasthan for different status
# (confirmed, recovered and deceased)
print(df.loc[:,['rj','status']].groupby('status').describe())




              rj                                                        
           count        mean         std  min   25%    50%    75%    max
status                                                                  
Confirmed  127.0  224.409449  191.507858  0.0  76.5  210.0  304.5  866.0
Deceased   127.0    4.354331    3.680768  0.0   1.0    4.0    7.0   17.0
Recovered  127.0  166.488189  155.543383  0.0  17.0  126.0  276.0  656.0
